targets/target_persistent.c

```diff
@@ -14,6 +21,39 @@ static int enumerate_ranges(const GumRangeDetails* details, gpointer user_data)
     return 0;
 }
 
+static void prefetch_log(const GumEvent* event, GumCpuContext* cpu_context, gpointer user_data) {
+    switch (event->type) {
+    case GUM_COMPILE: {
+        const GumCompileEvent* compile = &event->compile;
+        if (prefetch_compiled != NULL) {
+            g_hash_table_add(prefetch_compiled, compile->begin);
+        }
+    } break;
+    }
+}
+
+static void prefetch_write(int fd, GHashTable *table) {
+    GHashTableIter iter;
+    gpointer key, value;
+    g_hash_table_iter_init(&iter, table);
+    while (g_hash_table_iter_next(&iter, &key, &value)) {
+        int status = write(fd, &key, sizeof(key));
```

Comment:
I would consider using a shared memory IPC like AFL already uses for the coverage bitmap. This way, you should not need any syscalls and hence any world switches after the initial setup.